### PR TITLE
Fixed the Game Rule Failing Errors

### PR DESCRIPTION
**Pull request by noahklowden**
There was one small bug causing errors in the game rule, which has been fixed. A few more tests for game rules that should also be passing has been added. There were a few tests in game.spec that shouldn't be passing yet (relating to game start events, which will be implemented shortly) so I've also modified those. They should be failing currently, but that will change in the next code I push.

### CODE REVIEW

File: appGeo/src/tests/game.spec.ts [hunk 1]
```diff
@@ -201,9 +199,9 @@ QUnit.test("Game Start and endGame", function(assert) {
 
     const test_game = new Game(gameRules, test_map, player_map);
 
-    assert.equal(test_game.startGame(), SUCCESS, "Game starts with players");
-    assert.equal(test_game.getGameActive(), ACTIVE, "Started game is not active");
-    assert.equal(test_game.endGame(),SUCCESS, "Started game cannot endGame");
+    assert.equal(test_game.startGame(), FAILURE, "Game cannot start with too few players");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:202]: I think this test is still failing

File: appGeo/src/tests/game.spec.ts [hunk 2]
```diff
@@ -201,9 +199,9 @@ QUnit.test("Game Start and endGame", function(assert) {
...
+    assert.equal(test_game.getGameActive(), INACTIVE, "Started game is not active");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:203]: Also failing -- expected 0 returned 1

File: appGeo/src/tests/game.spec.ts [hunk 3]
```diff
@@ -201,9 +199,9 @@ QUnit.test("Game Start and endGame", function(assert) {
...
+    assert.equal(test_game.endGame(), FAILURE, "Unstarted game cannot endGame");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:204]: Failing -- expected -10 returned 10

**Comment by noahklowden** [comment]
As stated in PR, this test is intended to fail right now.

File: appGeo/src/tests/game.spec.ts [hunk 4]
```diff
@@ -221,8 +219,11 @@ QUnit.test("Game Start and endGame", function(assert) {
     assert.equal(test_game.getFractionRole(CIVILIAN), 0, "No role assigned, no civilians");
     assert.equal(test_game.getFractionRole(KILLER), 0, "No role assigned, no killers");
 
+    assert.equal(test_game.startGame(), SUCCESS, "Game can start when meeting minimum players");
+    assert.equal(test_game.getGameActive(), ACTIVE, "Started game is active")
+
     assert.equal(test_game.getPlayerCount(), 5, "Can count players in game");
-    assert.equal(test_game.getRoleCount(CIVILIAN), 4, "Should start game with 3 civilians");
+    assert.equal(test_game.getRoleCount(CIVILIAN), 4, "Should start game with 4 civilians");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:226]: Failing -- expected 4 returned 0

File: appGeo/src/tests/game.spec.ts [hunk 5]
```diff
@@ -221,8 +219,11 @@ QUnit.test("Game Start and endGame", function(assert) {
...
     assert.equal(test_game.getRoleCount(KILLER), 1, "Should start game with 1 killer");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:227]: Failing -- expected 1 returned 0

File: appGeo/src/tests/game.spec.ts [hunk 6]
```diff
@@ -221,8 +219,11 @@ QUnit.test("Game Start and endGame", function(assert) {
...
 
     assert.equal(test_game.getFractionRole(CIVILIAN), 0.8, "Should be 80% civilians");
```
Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:229]: Failing -- expected 0.8 returned 0

Comment (corian-jal) [inline appGeo/src/tests/game.spec.ts:229]: Fraction Role Civilian -- Expected 0.8 returned 0
Fraction Role Killer -- Expected 0.2 returned 0

**Review by corian-jal — APPROVED** [review]
All of the existing test failures will be addressed in later pull requests. As such, I think everything here looks good to be merged!

**Pull request merged**